webapp/src/ts/services/enketo-translation.service.ts

```diff
@@ -159,28 +159,30 @@ export class EnketoTranslationService {
    */
   contactRecordToJs(record) {
     const root = $.parseXML(record).firstChild;
-
     const result:any = {
       doc: null,
       siblings: {},
     };
+
     const repeats = this.repeatsToJs(root);
     if (repeats) {
       result.repeats = repeats;
     }
 
     const NODE_NAMES_TO_IGNORE = ['meta', 'inputs', 'repeat'];
 
-    this.withElements(root.childNodes)
-      .filter((node:any) => !NODE_NAMES_TO_IGNORE.includes(node.nodeName))
+    this
+      .withElements(root.childNodes)
+      .filter((node:any) => !NODE_NAMES_TO_IGNORE.includes(node.nodeName) && node.childElementCount > 0)
```

Comment:
This filters out 1st level fields and empty groups to only process groups with content to be docs.

Reply:
@latin-panda would it be a good idea to have a warning in `cht-conf` if someone tries to compile a contact form with 1st level fields that are not groups?

Reply:
🤔 I wonder if it's going to be helpful or annoying, depending how frequently app developers use top level fields for calculations inside groups vs how often this turns out to be a mistake/error in their forms.

Reply:
@latin-panda good point. I forgot that it is probably pretty common to just put calcs up at the top that might get referenced other places throughout the form.  Out of curiosity, what is happening to those calc values when the form is saved right now (before these changes)?  (I get from the described issue that they are not properly saved anywhere, but are they just dropped or are they thrown into some unreferenced doc?)